### PR TITLE
Add Integration tests

### PR DESCRIPTION
**Pull request by mtyazici**
Added Integrations-tests for the repo. On push a workflow will be triggered. A workflow can be triggered on Actions page also. It will do the following:

- ~~Clone the hazelcast main repo, build a jar~~ Download hazelcast jar version specified in `pom.xml`.
- Build a jar of hazelcast-azure
- Apply Terraform configuration to deploy two Hazelcast cluster members and Hazelcast Management Center on Azure.
- Verify that cluster is formed and members are able to communicate with Management Center.

If all these steps are done correctly the build will pass.

- [x] Required credentials should be added into repo as a secrets.

This worfklow will not run when a PR is created. This is because github doesn't have any way of providing secrets to a forked repository. So a workflow accessing the secrets will always fail.

A related [thread](https://github.community/t/make-secrets-available-to-builds-of-forks/16166/35) about this problem.

### CODE REVIEW

**Comment by mtyazici** [comment]
I applied suggestions from the code review in [AWS PR](https://github.com/hazelcast/hazelcast-aws/pull/196).

File: .github/terraform/scripts/start_azure_hazelcast_management_center.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    echo "Downloading JAR..."
+    if wget -q "$MANCENTER_JAR_URL"; then
+        echo "Hazelcast Management JAR downloaded successfully."
+    else
+        echo "Hazelcast Management JAR could NOT be downloaded!"
+        exit 1;
+    fi
+    unzip hazelcast-management-center-${MANCENTER_VERSION}.zip
+    cp -R hazelcast-management-center-${MANCENTER_VERSION}/* ./
+
```
Comment (enozcan) [inline .github/terraform/scripts/start_azure_hazelcast_management_center.sh:28]:
```suggestion
```

File: .github/terraform/scripts/start_azure_hazelcast_member.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+TAG_KEY=$1
+TAG_VALUE=$2
+
+sed -i -e "s/TAG_KEY/${TAG_KEY}/g" ${HOME}/hazelcast.yaml
+sed -i -e "s/TAG_VALUE/${TAG_VALUE}/g" ${HOME}/hazelcast.yaml
+
+CLASSPATH="${HOME}/jars/hazelcast.jar:${HOME}/jars/hazelcast-azure.jar:${HOME}/hazelcast.yaml"
+
+nohup java -cp ${CLASSPATH} -server com.hazelcast.core.server.HazelcastMemberStarter >> ${HOME}/logs/hazelcast.stderr.log 2>> ${HOME}/logs/hazelcast.stdout.log &
```
Comment (enozcan) [inline .github/terraform/scripts/start_azure_hazelcast_member.sh:13]: Applies the review from gcp:
```suggestion
nohup java -cp ${CLASSPATH} -server com.hazelcast.core.server.HazelcastMemberStarter &>> ${HOME}/logs/hazelcast.logs &
```

File: .github/terraform/main.tf [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+    host = azurerm_linux_virtual_machine.hazelcast_member[count.index].public_ip_address
+    timeout = "180s"
+    agent = false
+    private_key = file("${var.local_key_path}/${var.azure_key_name}")
+  }
+
+  provisioner "remote-exec" {
+    inline = [
+      "cd /home/${var.azure_ssh_user}",
+      "tail -n 20 ./logs/hazelcast.stdout.log",
```
Comment (enozcan) [inline .github/terraform/main.tf:207]: Regarding the change in start_member.sh:
```suggestion
      "tail -n 20 ./logs/hazelcast.logs",
```

**Review by enozcan — CHANGES_REQUESTED** [review]
Leaving a few comments from GCP reviews.

**Review by enozcan — APPROVED** [review]

**Review by hasancelik — APPROVED** [review]

**Pull request merged**